### PR TITLE
Support custom header

### PR DESCRIPTION
**Pull request by hcorta**
Hi! 
This PR extends @harajune proposal on [171](https://github.com/NotionX/react-notion-x/pull/171).
My main goal was enabling passing a custom header component, but allowing the information in the original to be accesible.

```
default function App() {

  function MyCustomHeader({
    activePageId,
    blockIds,
    blockMap,
    breadcrumbs,
    className,
    components,
    hasSearch,
    isSearchOpen,
    mapPageUrl,
    onCloseSearch,
    onOpenSearch,
    recordMap,
    rootPageId,
    searchNotion,
    setIsSearchOpen,
    ...rest
  }) {
    return <header>{'my custom header'}</header>
  }

  return (
      <NotionRenderer
         header={MyCustomHeader}
      />
  )
}
```

A few comments:
1. I have renamed the "page-header.tsx" to "header.tsx" (as well as the component) since the "pageHeader" prop refers to another component and it might leads to confusion. 
2. I have created an util for getting the breadcrumbs into the utils package -> "get-page-breadcrumbs.tsx"
3.  I have added the "disableHeader" prop to fully disable the header, if necessary.

As this is my first PR to the lib, I'd willing to hear suggestions on code style or similar.
🙂

### CODE REVIEW

File: packages/react-notion-x/src/components/header.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
+import React from 'react'
```
Comment (iNorthIsle) [inline packages/react-notion-x/src/components/header.tsx:1]: LGTM! But the PR will look better and less confused if change it in the `page-header` file while not creating a new one and deleting it 😁

**Comment by normdoow** [comment]
Thanks @hcorta will be looking at this over the next couple of days.

File: packages/react-notion-x/src/block.tsx [hunk 2]
```diff
@@ -32,13 +32,15 @@ interface BlockProps {
   className?: string
   bodyClassName?: string
 
+  header?: React.ElementType
   footer?: React.ReactNode
   pageHeader?: React.ReactNode
```
Comment (normdoow) [inline packages/react-notion-x/src/block.tsx:37]: aren't you replacing this pageHeader so it should be removed?

File: packages/react-notion-x/src/block.tsx [hunk 3]
```diff
@@ -66,12 +68,14 @@ export const Block: React.FC<BlockProps> = (props) => {
     level,
     className,
     bodyClassName,
+    header,
     footer,
     pageHeader,
```
Comment (normdoow) [inline packages/react-notion-x/src/block.tsx:73]: aren't you replacing this pageHeader so it should be removed?

File: packages/react-notion-x/src/components/header.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+import { getPageBreadcrumbs } from 'notion-utils'
+import { useNotionContext } from '../context'
+import { PageIcon } from './page-icon'
+import { SearchIcon } from '../icons/search-icon'
+import { cs } from '../utils'
+import { SearchDialog } from './search-dialog'
+
+export const Header: React.FC<{
+  header: React.ElementType
+}> = ({ header: CustomHeader }) => {
```
Comment (normdoow) [inline packages/react-notion-x/src/components/header.tsx:13]: Maybe I'm missing something but where is this CustomHeader type coming from? Obviously it requires some props.

**Comment by normdoow** [comment]
Just a few comments! Thanks! Looks pretty good though!

**Comment by hcorta** [comment]
@normdoow the pageHeader prop is not affected by my changes, and it is still is use, as it was, in **_<Block />_**. 
**_<CustomHeader/>_** is just the prop "**_header_**" that needs to be capitalised so it can to be used as a JSX element. I may rename it to <Header />, would that make it clearer?

**Comment by normdoow** [comment]
> @normdoow the **_pageHeader_** prop is not affected somehow by my changes, and it is still is use, as it was, in **_Block_**. **_CustomHeader_** is just the prop "**_header_**" that needs to be capitalised so it can to be used as a JSX element. I may rename it to **_Header_**, would that make it clearer?

No, CustomHeader is good. I just didn't see it imported from anywhere.

**Comment by hcorta** [comment]
@normdoow anything else here I can help with?

**Comment by normdoow** [comment]
All good!

**Comment by normdoow** [comment]
added to v4.12.0

**Comment by pbteja1998** [comment]
@normdoow @hcorta `PageHeader` is something different and I am using it. Any reason why it got removed here?
CleanShot 2021-12-18 at 07 59 35@2x


Page header previously used to be on top of the page title.

**Comment by normdoow** [comment]
> @normdoow @hcorta `PageHeader` is something different and I am using it. Any reason why it got removed here? CleanShot 2021-12-18 at 07 59 35@2x
> 
> Page header previously used to be on top of the page title.

Pageheader was renamed to header, so all that functionality should still be there.

**Comment by pbteja1998** [comment]
> > @normdoow @hcorta `PageHeader` is something different and I am using it. Any reason why it got removed here? CleanShot 2021-12-18 at 07 59 35@2x
> > Page header previously used to be on top of the page title.
> 
> Pageheader was renamed to header, so all that functionality should still be there.

Oh... I thought `header` was about replacing breadcrumbs.

**Comment by hcorta** [comment]
@pbteja1998 are you referring to the "pageHeader" prop, or the component "PageHeader" within the project?

The pageHeader prop and its use has not been changed at all.

The PageHeader component was not exposed previously, but now it is through the header prop, that allows you to pass a component and resolve any custom case, instead of having to do internal hacks.

Hope that helps☺️

**Pull request merged**